lib/stateTransition/AbstractStateTransition.js

```diff
@@ -49,6 +73,80 @@ class AbstractStateTransition {
   hash() {
     return hash(this.serialize()).toString('hex');
   }
+
+  /**
+   *  Returns signature
+   *
+   * @return {string|null}
+   */
+  getSignature() {
+    return this.signature;
+  }
+
+  /**
+   * Returns public key id
+   *
+   * @returns {number|null}
+   */
+  getPublicKeyId() {
+    return this.publicKeyId;
+  }
+
+  /**
+   * Returns user id
+   *
+   * @returns {string|null}
+   */
+  getUserId() {
+    return this.userId;
+  }
+
+  /**
+   * Sign data with privateKey
+   *
+   * @param {{id: number, userId: string, type: number, privateKey: string}} options
+   * @return {Buffer}
+   */
+  sign(options) {
```

Comment:
Accept PrivateKey, DPP PublicKey?

Reply:
The same as a comment above about public keys and verify method